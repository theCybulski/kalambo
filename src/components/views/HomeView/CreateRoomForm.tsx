import React from 'react';
import { useHistory } from 'react-router-dom';
import { observer } from 'mobx-react';
import { useStores } from 'hooks/useStores';
import axios from 'axios';
import Form from 'components/Form/Form';
import Input from 'components/Input/Input';
import Button from 'components/Button/Button';

import { API_BASE_ENDPOINT } from 'shared/config/config';
import socket from 'api/api';
import Heading, { headingVariant } from '../../Heading/Heading';

const CreateRoomForm: React.FC = observer(() => {
  const {
    playerStore: { setLocalPlayer },
    roomStore: { setRoomSettings },
  } = useStores();
  const history = useHistory();

  const handleCreateRoom = (data) => {
    const { name } = data;

    console.log(socket);
    axios({
      method: 'get',
      url: `${API_BASE_ENDPOINT}/api/v1/create_room?adminName=${name}&socketId=${socket.id}`,
    })
      .then(({ data }) => {
        if (data.error) return console.log(new Error(data.error));
        setLocalPlayer(data.player);
        setRoomSettings(data.room);

        history.push(`/room?id=${data.room.roomNo}`);
      })
      .catch((error) => console.log(error));
  };

  return (
    <>
      <Heading as={headingVariant.h2}>Create new room</Heading>
      <Form onSubmit={handleCreateRoom} resetOnSubmit>
        <Input
          name="name"
          placeholder="Your name"
          autoComplete="off"
          data-cy="input-name-create"
          isRequired
        />

        <Button type="submit" data-cy="btn-create-room">
          Create Room
        </Button>
      </Form>
    </>
  );
});

export default CreateRoomForm;
