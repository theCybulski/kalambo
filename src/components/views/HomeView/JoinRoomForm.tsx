import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { observer } from 'mobx-react';
import { useStores } from 'hooks/useStores';
import Form from 'components/Form/Form';
import Input from 'components/Input/Input';

import socket from 'api/api';
import Button from 'components/Button/Button';
import Heading, { headingVariant } from 'components/Heading/Heading';

const JoinRoomForm: React.FC = observer(() => {
  const {
    playerStore: { setLocalPlayer },
    roomStore: { setRoomSettings, joinRoom },
  } = useStores();
  const history = useHistory();

  const [error, setError] = useState(null);

  const handleJoinRoom = (data) => {
    const roomNo = data.roomNo.replace(/\s/g, '');
    joinRoom(data.name, socket.id, roomNo)
      .then(({ data }) => {
        if (data.error) {
          setError(data.error);
          return console.log(new Error(data.error));
        }

        setLocalPlayer(data.player);
        setRoomSettings(data.room);

        history.push(`/room?id=${data.room.roomNo}`);
      })
      .catch((error) => console.log(error));
  };

  return (
    <>
      <Heading as={headingVariant.h2}>Join existing room</Heading>
      <Form onSubmit={handleJoinRoom}>
        <Input name="name" placeholder="Your name" autoComplete="off" />
        <Input name="roomNo" placeholder="Room number" autoComplete="off" />

        {error && (
          <>
            {error}
            <br />
          </>
        )}

        <Button type="submit">Join Room</Button>
      </Form>
    </>
  );
});

export default JoinRoomForm;
