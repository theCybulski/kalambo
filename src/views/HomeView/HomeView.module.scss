.wrapper {
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 100vh;

  .formsWrapper {
    width: 300px;

    .formCreate {
      margin-bottom: 50px;
    }
  }
}
