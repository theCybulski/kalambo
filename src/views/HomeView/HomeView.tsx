import React from 'react';
import 'mobx-react-lite/batchingForReactDom';
import { observer } from 'mobx-react';

const HomeView: React.FC = observer(() => {
  const viewName = 'Home view';

  return (
    <div>
      <h1>{viewName}</h1>
    </div>
  );
});

export default HomeView;
