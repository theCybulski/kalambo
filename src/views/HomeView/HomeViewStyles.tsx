import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 100vh;
`;

export const FormsWrapper = styled.div`
  width: 300px;
`;
