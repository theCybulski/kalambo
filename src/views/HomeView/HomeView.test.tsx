import React from 'react';
import toJson from 'enzyme-to-json';

import { shallow } from 'enzyme';

import HomeView from './HomeView';

describe(`${HomeView.name}`, () => {
  it('Matches snapshot', () => {
    const wrapper = shallow(<HomeView />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
