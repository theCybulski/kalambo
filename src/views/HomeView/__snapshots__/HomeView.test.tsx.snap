// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`undefined Matches snapshot 1`] = `
<styled.div>
  <styled.div>
    <Memo(wrappedComponent) />
    <br />
    <br />
    <br />
    <Memo(wrappedComponent) />
  </styled.div>
</styled.div>
`;
